1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

First import pandas, Numpy, and the .csv files. Make a copy of the files for safekeeping.

In [1]:
import pandas as pd
import numpy as np
airports=pd.read_csv("C:/Users/sguzi/Desktop/airports.csv")
weather=pd.read_csv("C:/Users/sguzi/Desktop/weather.csv")
airports_copy = airports.copy()
weather_copy = weather.copy()

Latitude is north/south (y); Longitude is east/west (x). 
Need highest y to get northmost, need highest y to get eastmost. 
There was a /n value up at the top, plus according to google, northernmosot airport is BRW, not EEN.

In [74]:
northernmost = airports_copy.sort_values(by='lat', ascending=False)
#northernmost[northernmost.tzone.notnull()] nOpE
northernmost = northernmost[northernmost.tzone != '\\N']
northernmost.head(1)

faa                        name        lat         lon  alt   tz dst  \
230  BRW  Wiley Post Will Rogers Mem  71.285446 -156.766003   44 -9.0   A   

                 tzone  
230  America/Anchorage

As expected, an airport in Anchorage, Alaska's the northernmost one up there. Now for easternmost.
SO APPARENTLY THIS IS SO FAR WEST THAT IT'S EAST, BUT IT'S IN ASIA, BUT ALSO ANCHORAGE? #logic

In [69]:
easternmost = airports_copy.sort_values(by='lon', ascending=False)
easternmost.head(1)

faa          name        lat        lon  alt   tz dst              tzone
1290  SYA  Eareckson As  52.712275  174.11362   98 -9.0   A  America/Anchorage

Turns out that this airport is.... so West that it's actually... East? According to longitude values. Way to mess with my geographical standpoint on where Alaska is.

Let's filter by the date we need - February 12, 2013

In [23]:
mydate = weather_copy.query("month == 2 and day == 12")
#this was just for me to see that i filtered by the right date
#mydate.head()

Let's filter by the wind speed, so we can get the windiest location on Feb12

In [72]:
mydate.sort_values(by='wind_speed')
winds = mydate.tail()
winds.sort_values(by='wind_speed', ascending=False)
winds.head(1)

origin  year  month  day  hour   temp  dewp  humid  wind_dir  \
18445    LGA  2013      2   12    19  44.06  23.0  43.02     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
18445    20.71404  23.837303     0.0    1011.5   10.0  2013-02-12T19:00:00Z

The windiest airport is LGA at 19:00, with a wind speed of 20.7. I'm glad I wasn't travelling then.